src/Microsoft.AspNetCore.Sockets.Client.Http/HttpOptions.cs

```diff
@@ -3,17 +3,24 @@
 
 using System;
 using System.Collections.Generic;
+using System.Net;
 using System.Net.Http;
 using System.Net.WebSockets;
+using System.Security.Cryptography.X509Certificates;
 
 namespace Microsoft.AspNetCore.Sockets.Client.Http
 {
     public class HttpOptions
     {
-        public HttpMessageHandler HttpMessageHandler { get; set; }
+        public Func<HttpMessageHandler, HttpMessageHandler> HttpMessageHandler { get; set; }
         public IReadOnlyCollection<KeyValuePair<string, string>> Headers { get; set; }
         public Func<string> AccessTokenFactory { get; set; }
         public TimeSpan CloseTimeout { get; set; } = TimeSpan.FromSeconds(5);
+        public ICredentials Credentials { get; set; }
+        public X509CertificateCollection ClientCertificates { get; set; }
+        public CookieContainer Cookies { get; set; }
```

Comment:
`CookieContainer` or `IList<Cookie>`? Likewise `X509CertificateCollection` or `IList<X509Certificate>`?

Reply:
CookieContainer isn't a traditional collection data structure - https://msdn.microsoft.com/en-us/library/system.net.cookiecontainer%28v=vs.110%29.aspx

Reply:
I think we should use the same type as ClientWebSocketOptions and HttpClientHandler, even though they are concrete. Though we may want to pre-initialize them, which is what [`ClientWebSocketOptions` does](https://source.dot.net/#System.Net.WebSockets.Client/System/Net/WebSockets/ClientWebSocketOptions.cs,91) for example.